### DISCUSSION TITLE
Multithreaded suppliers

### DISCUSSION THREAD

**Discussion by kebp** [rdkit/rdkit#4990] [Q&A]
I'm having difficulties trying to get the MultithreadedSDMolSupplier to read a large SD file but am only able to get it to read the first molecule from the file. I'm using version 20121_09_4 and the very minimal code below for testing:

    from rdkit import Chem
    i = 0 
    with Chem.MultithreadedSDMolSupplier('./sample_100.sdf') as sdSupl:
        for mol in sdSupl:
            if mol is not None:
                i += 1
    print(i)

Any suggestions where I might be going wrong?

**Comment by bp-kelley** [comment]
I expect you may be reading more molecules, but most are not considered valid by the RDKit.

```
from rdkit import Chem
i = 0 
with Chem.MultithreadedSDMolSupplier('./sample_100.sdf') as sdSupl:
    for count, mol in enumerate(sdSupl):
        if mol is not None:
            i += 1
    print(f"Got {i} good molecules out of {count}")
```

**Reply by kebp** [reply]
Not convinced.  If I make the one change to your version substituting ForwardSDMolSupplier for MultithreadedSDMolSupplier the script returns "Got 100 good molecules out of 99" which suggests to me that all the molecules are considered valid by RDKit. Am I wrong and a file can be valid for a Forward supplier but not a Multithreaded one?

**Reply by bp-kelley** [reply]
Good sleuthing and thanks for the report.

**Comment by ptosco** [comment]
@kebp Thanks for reporting. Indeed `MultithreadedSDMolSupplier` bails out after the 1st molecule unless you set at least one property on them. I have filed this as #4992.
